lib/apis/v2/bgpconfig.go

```diff
@@ -37,11 +37,10 @@ type BGPConfiguration struct {
 	Spec BGPConfigurationSpec `json:"spec,omitempty"`
 }
 
-// TODO: Add validation on LogSeverityScreen for valid values
 // BGPConfigurationSpec contains the values of the BGP configuration.
 type BGPConfigurationSpec struct {
 	// LogSeverityScreen is the log severity above which logs are sent to the stdout. [Default: INFO]
-	LogSeverityScreen string `json:"logSeverityScreen,omitempty" validate:"omitempty" confignamev1:"loglevel"`
+	LogSeverityScreen string `json:"logSeverityScreen,omitempty" validate:"omitempty,bgpLogLevel" confignamev1:"loglevel"`
```

Comment:
I think we can get away with a single logLevel validator no?

Reply:
looks like bgp accepts a wider range of logging options:
`bgpLogLevel           = regexp.MustCompile("^(Info|Warning|Debug|Error|Fatal|Trace)$")`

versus:
`felixLogLevel         = regexp.MustCompile("^(Info|Warning|Debug|Error|Critical)$")`

unless Critical/Trace are incorrect, I'd say we need both regex's

Reply:
I think both Critical and Trace are incorrect.

They both use the libcalico-go logutils to set log level and that basically filters down to logrus.

Neither Critical nor Trace are options in the logrus logging levels.

I think Felix will basically handle CRITICAL as "Panic" but only on the grounds that it would not recognize Critical and then default to Panic (along with a warning log).

I suspect Felix needs updating to change CRITICAL to PANIC.

And the BGP one is just wrong